service/src/intTest/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/StorageConnectorTest.java

```diff
@@ -0,0 +1,56 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import uk.nhs.adaptors.gp2gp.extension.IntegrationTestsExtension;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@SpringBootTest
+@ExtendWith({ SpringExtension.class, IntegrationTestsExtension.class })
+public class StorageConnectorTest {
+    @Autowired
+    private StorageConnector storageConnector;
+
+    private static final String FILE_EXPECTED_DOWNLOAD = "src/intTest/resources/downloads/test.txt";
+    private static final String SETUP_FILE_TXT = "setupFile.txt";
+    private File uploadedFile;
+    private File downloadedFile;
+
+    @BeforeEach
+    public void setupConnector() throws IOException {
+        String fileToUpload = StorageConnectorTest.class.getResource("/test.txt").getPath();
+        uploadedFile = new File(fileToUpload);
```

Comment:
This instance method writes to a static field. This is tricky to get correct if multiple instances are being manipulated, and generally bad practice.